### DISCUSSION TITLE
GetCompletionValue() don't work in 3.0.1 version

### DISCUSSION THREAD

**Discussion by jricardogarcia** [sebastienros/jint#1811] [Q&A] [answered]
Hi,
Until version 2.11.58 I used function returns with .GetCompletionValue().
Updating to version 3.0.1, the return is no longer valid.
Below is it.

`
public object Executar(params object[] objetos)
{
    var engine = new Engine();
    var parametros = string.Join(",", _parametros);
    var valores = string.Join(",", objetos.Select(a => JsonConvert.SerializeObject(a)));

    var execute =
        $@"
            try{{
                execute({valores})
            }} catch(ex) {{ return ex;  }}
        ";

    var script = $"function execute({parametros}) {{{_formula}}}; {execute}";

    var valor = engine.Execute(script, new ParserOptions
    {
        Tolerant = true
    }).GetCompletionValue();

    return valor.Type switch
    {
        Types.Boolean => valor.AsBoolean(),
        Types.Number => valor.AsNumber(),
        Types.Object => valor.AsObject(),
        Types.String => valor.AsString(),
        _ => null,
    };
}
`

What would be the best way to update my code?

Thanks

**Comment by lahma** [comment, marked as answer]
Call `Engine.Evaluate` which returns a `JsValue` back.

**Reply by jricardogarcia** [reply]
Thank you very much
